src/strategies/information.rs

```diff
@@ -892,12 +895,17 @@ impl InformationPlayerStrategy {
         }
 
         // if anything is totally useless, discard it
-        if public_useless_indices.len() > 1 {
-            let info = public_info.get_hat_sum(public_useless_indices.len() as u32, view);
+        if public_useless_indices.len() > 0 {
+            let info_amt = public_useless_indices.len() as u32;
+            if let Some(private_useless_index) = private_useless_index {
+                let num_states = info_amt + 2;
+                if public_info.decide_action_not_known_to_be_possible(num_states, view) {
+                    return TurnChoice::Discard(*private_useless_index);
+                }
+            }
+            let info = public_info.get_hat_sum(info_amt, view);
```

Comment:
i think one problem is we're now this case even if public_useless_indices.len() == 1.  previously the idea is it's better to hold onto these discards since one you have 2 of them you get a free bit of info

Reply:
Ohh, I didn't consider that at all! A special case makes sense here (I think, but I'm not too sure) since it's the only case in which there aren't decreasing returns to having public useless cards: If we have privately known useless cards and discard, the number of bits we get is

| public useless cards | bits without this trick | bits with this trick (roughly) |
|------|--------|-------|
| 0 | 0 | 0 |
| 1 | 0 | 0.6 |
| 2 | 1 | 1.3 |
| 3 | 1.6 | 1.8 |
| 4 | 2 | 2.1 |

I'll report back if the special case ends up helping.

Reply:
Ok, the results are:

Reference (current tip unless I made a mistake):

|         |   2p    |   3p    |   4p    |   5p    |
|---------|------------------|------------------|------------------|------------------|
| cheat   | 24.8594 ± 0.0036 | 24.9785 ± 0.0012 | 24.9720 ± 0.0014 | 24.9557 ± 0.0018 |
|         | 90.59 ± 0.21 % | 98.17 ± 0.09 % | 97.76 ± 0.10 % | 96.42 ± 0.13 % |
| info    | 22.5239 ± 0.0125 | 24.7952 ± 0.0039 | 24.9353 ± 0.0022 | 24.9221 ± 0.0024 |
|         | 12.63 ± 0.23 % | 84.54 ± 0.26 % | 95.01 ± 0.15 % | 94.01 ± 0.17 % |

With `num_states = round(e * info_amt)`, without special case:

|         |   2p    |   3p    |   4p    |   5p    |
|---------|------------------|------------------|------------------|------------------|
| cheat   | 24.8594 ± 0.0036 | 24.9785 ± 0.0012 | 24.9720 ± 0.0014 | 24.9557 ± 0.0018 |
|         | 90.59 ± 0.21 % | 98.17 ± 0.09 % | 97.76 ± 0.10 % | 96.42 ± 0.13 % |
| info    | 22.5238 ± 0.0125 | 24.7950 ± 0.0039 | 24.9353 ± 0.0022 | 24.9221 ± 0.0024 |
|         | 12.62 ± 0.23 % | 84.54 ± 0.26 % | 95.01 ± 0.15 % | 94.02 ± 0.17 % |

With `num_states = round(e * info_amt)`, except if there's exactly one public useless card, in which case we discard the private one:

|         |   2p    |   3p    |   4p    |   5p    |
|---------|------------------|------------------|------------------|------------------|
| cheat   | 24.8594 ± 0.0036 | 24.9785 ± 0.0012 | 24.9720 ± 0.0014 | 24.9557 ± 0.0018 |
|         | 90.59 ± 0.21 % | 98.17 ± 0.09 % | 97.76 ± 0.10 % | 96.42 ± 0.13 % |
| info    | 22.5208 ± 0.0125 | 24.7945 ± 0.0039 | 24.9353 ± 0.0022 | 24.9221 ± 0.0024 |
|         | 12.58 ± 0.23 % | 84.47 ± 0.26 % | 95.02 ± 0.15 % | 94.01 ± 0.17 % |

So the special case doesn't appear to help, but I think the real takeaway is that the differences are all tiny, presumably because (I'd guess) privately, but not publicly known useless cards are just too rare to matter. So I'm totally fine with any version getting merged!

Reply:
cool, i think we should probably just not add it then, for now; given fixed performance, simpler strategies seem better :)